threedigrid/admin/serializers.py

```diff
@@ -135,6 +139,47 @@ def data(self):
             cls=NumpyEncoder,
         )
 
+    def _set_tabulated_cross_section_information(self, lines_data, i, properties):
+        """Add cross section information to line properties
+
+        Args:
+            line_data (Dict): lines data
+            i (int): index into the lines data
+            properties (Dict): to be exported line data
+        """
+        from threedigrid.admin.crosssections.models import CrossSections
+
+        if isinstance(self._coupled_model, CrossSections):
+            try:
+                cross1 = lines_data["cross1"][i]
+                (
+                    width,
+                    height,
+                ) = self._coupled_model.get_tabulated_cross_section_height_and_width(
+                    cross1
+                )
+                properties["cross_section_type_1"] = self._coupled_model.shape[cross1]
+                properties["cross_section_table_1"] = [width, height]
+                properties["cross_section_weight"] = lines_data["cross_weight"][i]
+            except IndexError:
+                properties["cross_section_type_1"] = None
+                properties["cross_section_table_1"] = [[], []]
+                properties["cross_section_weight"] = None
+
+            try:
+                cross2 = lines_data["cross2"][i]
+                (
+                    width,
+                    height,
+                ) = self._coupled_model.get_tabulated_cross_section_height_and_width(
+                    cross2
+                )
+                properties["cross_section_type_2"] = self._coupled_model.shape[cross2]
+                properties["cross_section_table_2"] = [width, height]
+            except IndexError:
+                properties["cross_section_type_2"] = None
+                properties["cross_section_table_2"] = [[], []]
```

Comment:
This is a personal preference, but I'd reduce the amount of duplicate code:

```suggestion
			for cross_idx in [1,2]:
				try:
					cross = lines_data[f"cross{cross_idx}"][i]
					(
						width,
						height,
					) = self._coupled_model.get_tabulated_cross_section_height_and_width(
						cross
					)
					properties["cross_section_type_{cross_idx}"] = self._coupled_model.shape[cross]
					properties["cross_section_table_{cross_idx}"] = [width, height]
					if cross_idx == 1:
						properties["cross_section_weight"] = lines_data["cross_weight"][i]
				except IndexError:
					properties[f"cross_section_type_{cross_idx}"] = None
					properties[f"cross_section_table_{cross_idx}"] = [[], []]
					if cross_idx == 1:                
						properties["cross_section_weight"] = None
```

Reply:
Had your solution first :) then switched to just 2 separate blocks becaue of readibilty, but both are good!